### PR TITLE
Adds RTC driver for Infineon XMC4xxx devices

### PR DESCRIPTION
**Pull request by talih0**
Adds support for settings/getting RTC time and using alarm/update feature.
    
The alarm option needs all fields to be set due to a hardware limitation.
    
RTC shares the same interrupt with the watchdog. Thus shared
interrupts must be enabled when WDT and RTC both need to trigger the ISR.

Devicetree for boards xmc45_relax_kit and xmc47_relax_kit are also updated to include RTC.

image

### CODE REVIEW

**Comment by talih0** [comment]
The compliance failure is a false positive. Though INTERUPT is misspelled it is referring to a function from Infineon hal.

File: drivers/rtc/rtc_xmc4xxx.c [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+	if (!XMC_RTC_IsRunning()) {
+		if (!XMC_SCU_HIB_IsHibernateDomainEnabled()) {
+			XMC_SCU_HIB_EnableHibernateDomain();
+		}
+
+		XMC_RTC_SetPrescaler(RTC_XMC4XXX_DEFAULT_PRESCALER);
+		XMC_RTC_SetTimeStdFormat(&stdtime);
+
+		XMC_RTC_Start();
+	}
```
Comment (bjarki-andreasen) [inline drivers/rtc/rtc_xmc4xxx.c:253]: The RTC should not be started and set to a random value at init, time is set manually using the rtc_set_time api :)

**Review by bjarki-andreasen — CHANGES_REQUESTED** [review]

**Comment by bjarki-andreasen** [comment]
Please run the alarm test suite using this PR #78739 it allows you to set the alarm time mask using Kconfig to match the rtc :)

**Comment by talih0** [comment]
> Please run the alarm test suite using this PR #73789 it allows you to set the alarm time mask using Kconfig to match the rtc :)

thanks, I rebased after the commit was merged:
image

**Review by bjarki-andreasen — CHANGES_REQUESTED** [review]
Could you split the PR into
1. add device driver
2. add rtc to boards
3. add boards to RTC test suite

It keeps bisectability high and makes reviews a bit easier :)

File: drivers/rtc/rtc_xmc4xxx.c [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+	irq_enable(DT_INST_IRQN(0));
+}
+#endif
+
+static int rtc_xmc4xxx_init(const struct device *dev)
+{
+	struct rtc_xmc4xxx_data *dev_data = dev->data;
+
+	if (XMC_RTC_IsRunning()) {
+		dev_data->is_time_set = true;
```
[the remark this answers is not in the document]
Comment (talih0) [reply drivers/rtc/rtc_xmc4xxx.c]: yep, thanks.

**Review by bjarki-andreasen — DISMISSED** [review]
Nice work!

**Review by ifyall — DISMISSED** [review]

**Comment by bjarki-andreasen** [comment]
Please ping me after feature freeze if I forget about this PR, we need to ping around a bit to get it merged with a CI failure ;)

**Comment by talih0** [comment]
@bjarki-andreasen, I removed the function name that was causing the compliance problem.

**Review by bjarki-andreasen — APPROVED** [review]

**Review by ifyall — APPROVED** [review]

**Pull request merged**